packages/backend-common/config.d.ts

```diff
@@ -72,10 +75,63 @@ export interface Config {
       /** Default database client to use */
       client: 'better-sqlite3' | 'sqlite3' | 'pg';
       /**
-       * Base database connection string or Knex object
+       * Base database connection string or Knex (PG) object
+       * PG object details come from: https://node-postgres.com/apis/client
        * @visibility secret
        */
-      connection: string | object;
+      connection:
+        | string
+        | {
```

Comment:
Could it work out to copy the work in [the other PR](https://github.com/backstage/backstage/pull/17852/files#diff-fe51777b408df4147da1051148208fc9dac33f74d4f9085be53bc7aa98bbf690R45)? Ie making this a

```ts
Partial<{
  /** @secret */
  thingToHide: string;
  [key: string]: unknown;
)>
```

Reply:
Yes, that works and is much nicer. I've updated the code, the changeset, and the PR description and title to reflect this.